packages/functions/src/service.ts

```diff
@@ -336,3 +351,168 @@ async function callAtURL(
+  const context = await functionsInstance.contextProvider.getContext();
+  if (context.authToken) {
+    headers['Authorization'] = 'Bearer ' + context.authToken;
+  }
+  if (context.messagingToken) {
+    headers['Firebase-Instance-ID-Token'] = context.messagingToken;
+  }
+  if (context.appCheckToken !== null) {
+    headers['X-Firebase-AppCheck'] = context.appCheckToken;
+  }
+
+  headers['Content-Type'] = 'application/json';
+  headers['Accept'] = 'text/event-stream';
+
+  let response: Response;
+  try {
+    response = await functionsInstance.fetchImpl(url, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      headers,
+      signal: options?.signal
+    });
+  } catch (e) {
+    if (e instanceof Error && e.name === 'AbortError') {
+      const error = new FunctionsError(
+        'cancelled',
+        'Request was cancelled.'
+      );
+      return {
+        data: Promise.reject(error),
+        stream: {
+          [Symbol.asyncIterator]() {
+            return {
+              next() {
+                return Promise.reject(error);
+              }
+            };
+          }
+        }
+      };
+    }
+    // This could be an unhandled error on the backend, or it could be a
+    // network error. There's no way to know, since an unhandled error on the
+    // backend will fail to set the proper CORS header, and thus will be
+    // treated as a network error by fetch.
+    const error = _errorForResponse(0, null)
+    return {
+      data: Promise.reject(error),
+      // Return an empty async iterator
+      stream: {
+        [Symbol.asyncIterator]() {
+          return {
+            next() {
+              return Promise.reject(error);
+            }
+          };
+        }
+      }
+    };
+  }
+
+  const reader = response.body!.getReader();
+  const decoder = new TextDecoder();
+
+  let buffer = '';
+  let resultResolver: (value: unknown) => void;
+  let resultRejecter: (reason: any) => void;
+
+  const resultPromise = new Promise<unknown>((resolve, reject) => {
+    resultResolver = resolve;
+    resultRejecter = reject;
+  });
+
+  options?.signal?.addEventListener('abort', () => {
+    reader.cancel();
+    const error = new FunctionsError(
+      'cancelled',
+      'Request was cancelled.'
+    );
+    resultRejecter(error);
+  });
+
+  const stream = {
+    [Symbol.asyncIterator]() {
+      return {
+        async next() {
+          if (options?.signal?.aborted) {
+            const error = new FunctionsError(
+              'cancelled',
+              'Request was cancelled.'
+            );
+            resultRejecter(error)
+            throw error;
+          }
+
+          while (true) {
+            const { value, done } = await reader.read();
+            if (done) return { done: true, value: undefined };
+
+            buffer += decoder.decode(value, { stream: true });
+            const lines = buffer.split('\n');
+            buffer = lines.pop() || '';
```

Comment:
this logic doesn't work if the server happens to return multiple data lines in a single chunk

Reply:
fixed.